scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala

```diff
@@ -381,10 +383,18 @@ class AstNamerMacros(val c: Context) extends Reflection with CommonNamerMacros {
           }
           if (params.nonEmpty) {
             val latestTree = getUnapply(params)
-            mstats1 += paramsVersions.headOption.fold {
-              latestTree
-            } { ver =>
-              getUnapply(paramsForVersion(ver), getDeprecatedAnno(ver))
+            mstatsPerVersion match {
+              case (headVer, headMstats) :: tail =>
+                val headParams = paramsForVersion(headVer)
+                headMstats += getUnapply(headParams)
+                tail.foreach { case (ver, verMstats) =>
+                  verMstats += getUnapply(paramsForVersion(ver))
+                }
+                val afterLastVer = getAfterVersion(mstatsPerVersion.last._1)
+                val anno = getDeprecatedAnno(headVer, s"; use `.$afterLastVer`")
```

Comment:
isn't this the "since" version we are advertising? (see your comment before next to `def getDeprecatedAnno`)